src/models/Shape.js

```diff
@@ -24,6 +26,18 @@ export default class Shape {
     return shapeOptions
   }
 
+  getVerticesWithCache(state) {
+    this.cache.forEach( (cachedShape) => {
+      var [shape, vertices] = cachedShape
+      if (JSON.stringify(state.shape) === JSON.stringify(shape)) {
```

Comment:
The Internet said this was fragile, but the other suggestions looked terrible. Is this the right way to compare the state.shape objects?